x-pack/platform/plugins/shared/security/server/uiam/uiam_service.ts

```diff
@@ -157,6 +225,78 @@ export class UiamService implements UiamServicePublic {
     }
   }
 
+  /**
+   * See {@link UiamServicePublic.grantApiKey}.
+   */
+  async grantApiKey(authorization: HTTPAuthorizationHeader, params: GrantUiamAPIKeyParams) {
+    try {
+      this.#logger.debug(
+        `Attempting to grant API key using authorization scheme: ${authorization.scheme}`
+      );
+
+      const body: GrantUiamApiKeyRequestBody = {
```

Comment:
From: https://github.com/elastic/kibana/pull/241050#discussion_r2549744890

> nit: it might be beneficial to create an interface (or rather a set of UIAM-centric interfaces) that describe (in types and JSDoc) the requests and responses for these endpoints. This will help us to adhere to the interfaces, catch type errors and also make it clear for the readers what every field means

@azasypkin Definitely cleaner. See above for interface definitions